### PR TITLE
UtxoIndex -> C.UTxO, TxId -> C.TxId, TxOutRef -> C.TxIn

### PR DESCRIPTION
**Pull request by sjoerdvisscher**
Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Formatting, PNG optimization, etc. are updated
    - [x] Important changes are reflected in changelog.d of the affected packages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-ledger/changelog.d/20230412_134023_sjoerd.visscher_PLT_5294_replace_more_plutus_types.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+### Removed
+
+- Removed `TxIn` and `TxInput` types.
```
Comment (berewt) [inline plutus-ledger/changelog.d/20230412_134023_sjoerd.visscher_PLT_5294_replace_more_plutus_types.md:3]: Maybe explain that we use Cardano types instead?

**Review by berewt — APPROVED** [review]
It looks a lot simpler now!

Just a minor remark about the changeling:

File: plutus-ledger/changelog.d/20230412_134023_sjoerd.visscher_PLT_5294_replace_more_plutus_types.md (hunk 1 above)
Comment (sjoerdvisscher) [reply plutus-ledger/changelog.d/20230412_134023_sjoerd.visscher_PLT_5294_replace_more_plutus_types.md:3]: That's not true actually. `TxInput` wasn't used at all anymore and the only place where `TxIn` was still used was in `ChainIndexTx` where I replaced it with `TxOutRef`. (The chain-index leans heavily on TxOutRef, and I have not replaced it there.)

File: plutus-contract/src/Plutus/Trace/Emulator/ContractInstance.hs [hunk 2]
```diff
@@ -345,11 +346,11 @@ updateTxOutStatus txns = do
     -- Check whether the contract instance is waiting for a status change of a
     -- transaction output of any of the new transactions. If that is the case,
     -- call 'addResponse' to sent the response.
-    let getSpentOutputs = map txInRef . view citxInputs
+    let getSpentOutputs = view citxInputs
         txWithTxOutStatus tx = let validity = validityFromChainIndex tx in
              fmap (, Committed validity (Spent (_citxTxId tx))) (getSpentOutputs tx)
           <> fmap (, Committed validity Unspent) (txOutRefs tx)
-        statusMap = Map.fromList $ foldMap txWithTxOutStatus txns
+        statusMap = Map.fromList $ either (error . show) id $ traverse (\(ref, a) -> (, a) <$> toCardanoTxIn ref) $ foldMap txWithTxOutStatus txns
```
Comment (ak3n) [inline plutus-contract/src/Plutus/Trace/Emulator/ContractInstance.hs:353]: I think it's good to add the context here. "getSpentOutputs: " for example.

File: plutus-contract/src/Plutus/Trace/Emulator/ContractInstance.hs [hunk 3]
```diff
@@ -532,6 +533,6 @@ indexBlock :: [ChainIndexTx] -> IndexedBlock
 indexBlock = foldMap indexTx where
   indexTx otx =
     IndexedBlock
-      { ibUtxoSpent = Map.fromSet (const otx) $ Set.fromList $ map txInRef $ view citxInputs otx
+      { ibUtxoSpent = Map.fromSet (const otx) $ Set.fromList $ either (error . show) id $ traverse toCardanoTxIn $ view citxInputs otx
```
Comment (ak3n) [inline plutus-contract/src/Plutus/Trace/Emulator/ContractInstance.hs:536]: Same here.

File: plutus-pab/src/Plutus/PAB/Arbitrary.hs [hunk 4]
```diff
@@ -211,6 +196,9 @@ instance Arbitrary Slot where
     shrink = genericShrink
 
 instance Arbitrary TxId where
+    arbitrary = pure ""
```
Comment (ak3n) [inline plutus-pab/src/Plutus/PAB/Arbitrary.hs:199]: We can't reuse the cardano's instance?

If not — maybe return some constant? The genesis TxId for example?

**Review by ak3n — APPROVED** [review]
LGTM, good work 👍

**Comment by sjoerdvisscher** [comment]
@koslambrou Another failed mac build, can you merge this?

**Pull request merged**